### PR TITLE
coll/libnbc: revisit NBC_Handle usage

### PR DESCRIPTION
**Pull request by ggouaillardet**
make NBC_Handle (almost) an internal structure created
by NBC_Schedule_request()
use a local variable instead of what was previously handle->tmpbuf

Refs open-mpi/ompi#3487

Signed-off-by: Gilles Gouaillardet <gilles@rist.or.jp>
(cherry picked from commit 9ba85b85e1fdc9cb6860a48d74f5c7216cbe42cd)

### CODE REVIEW

**Comment by bwbarrett** [comment]
@ggouaillardet , can you rebase and clean up the merge?

**Comment by ggouaillardet** [comment]
@bwbarrett done, that was just a trivial conflict of copyrights

File: ompi/mca/coll/libnbc/nbc_ireduce.c [hunk 1]
```diff
@@ -343,14 +328,15 @@ static inline int red_sched_binomial (int rank, int p, int root, const void *sen
       vpeer = vrank + (1 << (r - 1));
       VRANK2RANK(peer, vpeer, vroot)
       if (peer < p) {
+        int tbuf;
```
Comment (hppritcha) [inline ompi/mca/coll/libnbc/nbc_ireduce.c]: what is tbuf being used for?  maybe github diff is confusing me?

Comment (hppritcha) [reply ompi/mca/coll/libnbc/nbc_ireduce.c]: oh never mind see it now.

**Pull request merged**